src/core/unit.c

```diff
@@ -516,6 +516,7 @@ void unit_free(Unit *u) {
         if (u->cgroup_path) {
                 hashmap_remove(u->manager->cgroup_unit, u->cgroup_path);
                 free(u->cgroup_path);
+                u->cgroup_path = NULL;
```

Comment:
If we wanted to follow upstream "[more closely](https://github.com/systemd/systemd/blob/master/src/core/cgroup.c#L2402-L2405)" we could go with `u->cgroup_path = mfree(u->cgroup_path);` but otherwise it's exactly as how it was resolved there :-)

Reply:
There are some other places where `free(u->cgroup_path); u->cgroup_path = NULL;` is used.
If you prefer, I can do the change in this PR.